File: Object-Oriented Programming/Python/Exercises/Ex1.py
print("Digite um valor para se calcular a raiz quadrada: ")
x = input()

print("Digite um valor como chute inicial: ")
x0 = input()

xi = x0
error_out_of_range = True

while error_out_of_range == True:
    prev_xi = xi
    xi = (float (prev_xi) + ((float (x) / float(prev_xi)))) / 2

    if (float (xi) - float (prev_xi)) <= 0.000000001:
        prev_xi = xi
        xi = (float (prev_xi) + ((float (x) / float(prev_xi)))) / 2
        error_out_of_range = False


print("A raiz quadrada de {:.7f} é: {:.7f}\n".format(float(x), float (xi)))

File: Object-Oriented Programming/Java/Exercises/src/Ex1.java

public class Ex1 {

	public static void main(String[] args) throws Exception {
		System.out.printf("Digite um double para se calcular a raiz: ");
		double x = EntradaTeclado.leDouble();
		
		System.out.printf("Digite um double como chute inicial: ");
		double x0 = EntradaTeclado.leDouble();
		
		double xi = x0, prev_xi = x0;
		
		do {
			prev_xi = xi;
			xi = (prev_xi + (x/prev_xi))/2;
	
		} while(xi - prev_xi >= 0.00000001);
			
		System.out.printf("A raiz quadrada de %f é: %f\n", x, xi);
	}
}
